src/core/server/opensearch/opensearch_config.ts

```diff
@@ -135,7 +135,24 @@ export const configSchema = schema.object({
   ),
 });
 
-const deprecations: ConfigDeprecationProvider = () => [
+const deprecations: ConfigDeprecationProvider = ({ renameFromRoot }) => [
+  renameFromRoot('elasticsearch.sniffOnStart', 'opensearch.sniffOnStart'),
+  renameFromRoot('elasticsearch.sniffInterval', 'opensearch.sniffInterval'),
+  renameFromRoot('elasticsearch.sniffOnConnectionFault', 'opensearch.sniffOnConnectionFault'),
+  renameFromRoot('elasticsearch.hosts', 'opensearch.hosts'),
+  renameFromRoot('elasticsearch.username', 'opensearch.username'),
+  renameFromRoot('elasticsearch.password', 'opensearch.password'),
+  renameFromRoot('elasticsearch.requestHeadersWhitelist', 'opensearch.requestHeadersWhitelist'),
+  renameFromRoot('elasticsearch.customHeaders', 'opensearch.customHeaders'),
+  renameFromRoot('elasticsearch.shardTimeout', 'opensearch.shardTimeout'),
+  renameFromRoot('elasticsearch.requestTimeout', 'opensearch.requestTimeout'),
+  renameFromRoot('elasticsearch.pingTimeout', 'opensearch.pingTimeout'),
+  renameFromRoot('elasticsearch.logQueries', 'opensearch.logQueries'),
+  renameFromRoot('elasticsearch.optimizedHealthcheckId', 'opensearch.optimizedHealthcheckId'),
+  renameFromRoot('elasticsearch.ssl', 'opensearch.ssl'),
+  renameFromRoot('elasticsearch.apiVersion', 'opensearch.apiVersion'),
+  renameFromRoot('elasticsearch.healthCheck', 'opensearch.healthCheck'),
+  renameFromRoot('elasticsearch.ignoreVersionMismatch', 'opensearch.ignoreVersionMismatch'),
```

Comment:
Will these need to be updated any time we introduce a new setting?

Reply:
When we add a new setting than probably not, I added your `optimizedHealthcheckId` just thinking if somebody saw their legacy config and thought since there are so many settings for engines that somebody might accidentally add that settings there. I can remove it though since it is completely new. Let me know what you think.